src/main/resources/vm/script/datamanagement/downloadFunctions.vm

```diff
@@ -94,6 +94,41 @@ function downloadGirderFile {
 }
 export -f downloadGirderFile
 
+#
+# URI are of the form of the following example.  A single "/", instead
+# of 3, after "shanoir:" is also allowed.
+# shanoir:/download.dcm?apiurl=https://shanoir-ng-nginx/shanoir-ng/datasets/carmin-data/path&format=dcm&datasetId=1&token=sqdjuzadj....&refreshToken=eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lk....
+# 
+# This method depends on refresh token process to refresh the token when it needs
+#
+function downloadShanoirFile {
+    local URI=$1
+    
+    wait_for_token
+
+    local token=`cat $shanoir_token_location`
+
+    echo "token inside download : ${token}"
+
+    local fileName=`echo $URI | sed -r 's#^shanoir:/(//)?([^/].*)\?.*$#\2#i'`
+    local apiUrl=`echo $URI | sed -r 's/^.*[?&]apiurl=([^&]*)(&.*)?$/\1/i'`
+    local format=`echo $URI | sed -r 's/^.*[?&]format=([^&]*)(&.*)?$/\1/i'`
+    local datasetId=`echo $URI | sed -r 's/^.*[?&]datasetId=([^&]*)(&.*)?$/\1/i'`
+
+    COMMAND(){ 
+        curl --write-out '%{http_code}' -o ${fileName} --request GET "${apiUrl}/${datasetId}?format=${format}" --header "Authorization: Bearer ${token}" --cacert /var/www/prod/shanoir-ng-nginx.pem
```

Comment:
The `cacert` argument should not be here.
I think you add the certificate in the standard certificate store of the moteur machine.